src/Simulation/DepletionVoltage.jl

```diff
@@ -0,0 +1,163 @@
+# """
+function _adapt_weighting_potential_to_electric_potential_grid!(sim::Simulation, contact_id::Int)
+    @assert !ismissing(sim.electric_potential) "Electric potential missing"
+    if ismissing(sim.weighting_potentials[contact_id]) calculate_weighting_potential!(sim, contact_id, verbose = false) end
+    if sim.weighting_potentials[contact_id].grid != sim.electric_potential.grid
+        sim.weighting_potentials[contact_id] = sim.weighting_potentials[contact_id][sim.electric_potential.grid]
+        #apply_initial_state!(sim, WeightingPotential, contact_id, sim.electric_potential.grid)
+        update_till_convergence!(sim, WeightingPotential, contact_id)
+    end
+end
+
+"""
+    get_depletion_voltage( sim::Simulation{T}, contact_id::Int, potential_range::AbstractRange; kwargs... )::T
+
+Returns the potential (in V) needed at the [`Contact`](@ref) with id `contact_id`
+to fully deplete the detector in a given [`Simulation`](@ref). For this, all other
+contact potentials are set to `0` and the potential at the specified contact is
+increased or decreased according to the `potential_range`. The depletion voltage
+is set to the potential for which a previously undepleted detector becomes depleted.
+
+## Arguments 
+* `sim::Simulation{T}`: [`Simulation`](@ref) for which the depletion voltage should be determined.
+* `contact_id::Int`: The `id` of the [`Contact`](@ref) at which the potential is applied.
+* `potential_range::AbstractRange`: Range of potentials to be tested. Must be strictly positive or negative.
+    
+## Keywords
+* `verbose::Bool = true`: Activate or deactivate additional info output. Default is `true`.
+
+## Example 
+```julia
+using SolidStateDetectors
+sim = Simulation(SSD_examples[:InvertedCoax])
+calculate_electric_potential!(sim)
+get_depletion_voltage(sim, 2, 1600:1:2500)
+```
+
+!!! warn
+    This method only works if the initial `sim` was calculated for a case in which the detector is fully depleted.
+
+!!! note
+    The accuracy of the result depends on the precision of the initial simulation.
+    
+See also [`is_depleted`](@ref).
+"""
+function get_depletion_voltage(sim::Simulation{T}, contact_id::Int,
+            potential_range::AbstractRange = range(extrema(broadcast(c -> c.potential, sim.detector.contacts))..., length = 1001);
+            verbose = true)::T where {T <: AbstractFloat}
+    
+    @assert is_depleted(sim.point_types) "This method only works for fully depleted simulations. Please increase the potentials in the configuration file to a greater value."
+    @assert first(potential_range) * last(potential_range) >= 0 "Please search for depletion voltage either in a fully positive or a fully negative range. Currently you were looking in the range
+        from $(first(potential_range)) to $(last(potential_range))"
+    potential_range = (last(potential_range) > 0 && first(potential_range) >= 0) || (step(potential_range) < 0) ? potential_range : reverse(potential_range)
```

Comment:
As commented before, I do not see why we should limit the `potential_range` to an increasing positive one or a decreasing negative one. We might wanna compare the execution speeds though and add a `@warn` or `@info` what is preferred (I guess from undepleted to depleted as the check for undepleted points is now ended once one point is undepleted). Also, it might make sense to check the depletion status at the beginning and at the end of `potential_range` and demand that it changes such that we can be sure that the expected depletion voltage is within that range.